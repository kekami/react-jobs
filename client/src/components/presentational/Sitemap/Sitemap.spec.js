import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Sitemap } from './Sitemap';


describe('Sitemap Component', () => {
  it('Sitemap (and children) renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Sitemap />
      </MemoryRouter>, div);
  });

  it('Sitemap renders correclty/ shallow test render', () => {
    const wrapper = shallow(<Sitemap />);
  });

  it('Sitemap matches its shallow snapshot', () => {
    const wrapper = shallow(<Sitemap />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});
