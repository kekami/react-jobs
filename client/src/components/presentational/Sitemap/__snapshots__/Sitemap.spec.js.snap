// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sitemap Component Sitemap matches its Full DOM snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "e0g5oz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "e0g5oz",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Sitemap>
      <styled.div>
        <div
          className="sc-bdVaJa gAkbBV"
        >
          <styled.div>
            <div
              className="sc-bwzfXH cuvyJy"
            >
              <styled.header>
                <header
                  className="sc-htpNat hKtJNh"
                >
                  <strong>
                    FOR TALENT
                  </strong>
                </header>
              </styled.header>
              <styled.div>
                <div
                  className="sc-bxivhb kklqTC"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      HOME
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/jobs"
                  >
                    <a
                      href="/jobs"
                      onClick={[Function]}
                    >
                      EXPLORE JOBS
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/profile"
                  >
                    <a
                      href="/profile"
                      onClick={[Function]}
                    >
                      MY PROFILE
                    </a>
                  </Link>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      SAVED JOBS
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      MY APPLICATIONS
                    </p>
                  </styled.p>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="sc-bwzfXH cuvyJy"
            >
              <styled.header>
                <header
                  className="sc-htpNat hKtJNh"
                >
                  <strong>
                    FOR BUSINESS
                  </strong>
                </header>
              </styled.header>
              <styled.div>
                <div
                  className="sc-bxivhb kklqTC"
                >
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      WHAT WE OFFER
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      PRICING
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      CUSTOMERS
                    </p>
                  </styled.p>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="sc-bwzfXH cuvyJy"
            >
              <styled.header>
                <header
                  className="sc-htpNat hKtJNh"
                >
                  <strong>
                    ABOUT US
                  </strong>
                </header>
              </styled.header>
              <styled.div>
                <div
                  className="sc-bxivhb kklqTC"
                >
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      OUR STORY
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      SUPPORT
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      JOIN THE TEAM
                    </p>
                  </styled.p>
                  <styled.p>
                    <p
                      className="sc-ifAKCX kbgPCA"
                    >
                      BLOG
                    </p>
                  </styled.p>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="sc-bwzfXH cuvyJy"
            >
              <styled.header>
                <header
                  className="sc-htpNat hKtJNh"
                >
                  <strong>
                    FOLLOW US
                  </strong>
                </header>
              </styled.header>
              <styled.div>
                <div
                  className="sc-bxivhb kklqTC"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <div>
                        FACEBOOK
                        <styled.i
                          aria-hidden="true"
                          className="fa fa-facebook-square fa-2x"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-facebook-square fa-2x sc-EHOje usOfa"
                          />
                        </styled.i>
                      </div>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <div>
                        TWITTER
                        <styled.i
                          aria-hidden="true"
                          className="fa fa-twitter-square fa-2x"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-twitter-square fa-2x sc-gzVnrw kputPq"
                          />
                        </styled.i>
                      </div>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <div>
                        LINKEDIN
                        <styled.i
                          aria-hidden="true"
                          className="fa fa-linkedin-square fa-2x"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-linkedin-square fa-2x sc-bZQynM dOeZDi"
                          />
                        </styled.i>
                      </div>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <div>
                        GOOGLE+
                        <styled.i
                          aria-hidden="true"
                          className="fa fa-google-plus-square fa-2x"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-google-plus-square fa-2x sc-htoDjs cJAawD"
                          />
                        </styled.i>
                      </div>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <div>
                        INSTAGRAM
                        <styled.span>
                          <span
                            className="sc-dnqmqq jHoYXs"
                          >
                            <styled.i
                              aria-hidden="true"
                              className="fa fa-instagram fa-2x"
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-instagram fa-2x sc-iwsKbI jEQMoS"
                              />
                            </styled.i>
                          </span>
                        </styled.span>
                      </div>
                    </a>
                  </Link>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Sitemap>
  </Router>
</MemoryRouter>
`;

exports[`Sitemap Component Sitemap matches its shallow snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.header>
      <strong>
        FOR TALENT
      </strong>
    </styled.header>
    <styled.div>
      <Link
        replace={false}
        to="/"
      >
        HOME
      </Link>
      <Link
        replace={false}
        to="/jobs"
      >
        EXPLORE JOBS
      </Link>
      <Link
        replace={false}
        to="/profile"
      >
        MY PROFILE
      </Link>
      <styled.p>
        SAVED JOBS
      </styled.p>
      <styled.p>
        MY APPLICATIONS
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.header>
      <strong>
        FOR BUSINESS
      </strong>
    </styled.header>
    <styled.div>
      <styled.p>
        WHAT WE OFFER
      </styled.p>
      <styled.p>
        PRICING
      </styled.p>
      <styled.p>
        CUSTOMERS
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.header>
      <strong>
        ABOUT US
      </strong>
    </styled.header>
    <styled.div>
      <styled.p>
        OUR STORY
      </styled.p>
      <styled.p>
        SUPPORT
      </styled.p>
      <styled.p>
        JOIN THE TEAM
      </styled.p>
      <styled.p>
        BLOG
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.header>
      <strong>
        FOLLOW US
      </strong>
    </styled.header>
    <styled.div>
      <Link
        replace={false}
        to="/"
      >
        <div>
          FACEBOOK
          <styled.i
            aria-hidden="true"
            className="fa fa-facebook-square fa-2x"
          />
        </div>
      </Link>
      <Link
        replace={false}
        to="/"
      >
        <div>
          TWITTER
          <styled.i
            aria-hidden="true"
            className="fa fa-twitter-square fa-2x"
          />
        </div>
      </Link>
      <Link
        replace={false}
        to="/"
      >
        <div>
          LINKEDIN
          <styled.i
            aria-hidden="true"
            className="fa fa-linkedin-square fa-2x"
          />
        </div>
      </Link>
      <Link
        replace={false}
        to="/"
      >
        <div>
          GOOGLE+
          <styled.i
            aria-hidden="true"
            className="fa fa-google-plus-square fa-2x"
          />
        </div>
      </Link>
      <Link
        replace={false}
        to="/"
      >
        <div>
          INSTAGRAM
          <styled.span>
            <styled.i
              aria-hidden="true"
              className="fa fa-instagram fa-2x"
            />
          </styled.span>
        </div>
      </Link>
    </styled.div>
  </styled.div>
</styled.div>
`;
